### PR TITLE
Build one package per service

### PR DESCRIPTION
**Pull request by ben-toogood**
Build a single GitHub package per service when it changes.

### CODE REVIEW

**Comment by ben-toogood** [comment]
Do not merge, need to apply a backend fix for GitHub API

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,61 +28,36 @@ jobs:
...
         echo ${{ secrets.GITHUB_TOKEN }} | docker login -u ${{ github.actor }} --password-stdin docker.pkg.github.com
-        docker build ~/builds -f Dockerfile -t $tag
-        docker push $tag
-        rm /home/runner/.docker/config.json # Delete credentials
+
+        services=(${{ steps.services_changed.outputs.services }}) 
+        for srv in "${services[@]}"; do
+          cp $srv/$srv tmp/app
+          tag=docker.pkg.github.com/${{ github.repository }}/$srv
+          docker build tmp -f Dockerfile -t $tag
```
Comment (asim) [inline .github/workflows/build.yml]: Moved to `.github/workflows/Dockerfile` to separate out the open source "go run" flow from the internal platform flow.

**Comment by asim** [comment]
I've merged master to fixup some breaks.

Made one change. I've moved the github actions specific Dockerfile to .github/workflows/Dockerfile. Was suffering from a break at the weekend as the docker image pushed to dockerhub didn't work since the top level Dockerfile changed. I want to keep the existing top level Dockerfile working with entrypoint.sh (the go run experience) and keep our github actions workflow Dockerfile in the workflow dir. Where people are using the open source it will use go run (for now) and we can focus on iterating on github actions internally until its in a good working state. I'm thinking more broadly about the [Network](https://micro.mu/docs/network.html) and the fact that we want systems to pull and run source from anywhere without worrying about builds (we may end up enforcing verified builds so this might change). I don't think this will cause any issues and its probably a good idea to keep the workflow stuff self contained for the time being.

We'll need to figure out how the "per service source" will work given the current go-micro implementation. We should be able to pass through the per service "Source" in the CreateOption when calling runtime.Create. Then process the app command as the binary name as expected. What this would potentially allow us to do is build a service's binaries for srv, api and web in the one container, maintaining a domain boundary identical to the directory structure in the mono repo. This would also allow `micro run helloworld/api` or `micro run helloworld/web` to work as is currently possible.

**Comment by asim** [comment]
Sorry, also force pushed to fix something because a binary was commited to master and needed to remove it.

**Comment by asim** [comment]
Merge at will!

**Pull request merged**